solutions/swb-reference/src/services/dataSetService.test.ts

```diff
@@ -684,4 +684,207 @@ describe('DataSetService', () => {
       expect(mockWorkbenchDataSetService.getAllDataSetAccessPermissions).toHaveBeenCalled();
     });
   });
+
+  describe('listDataSetsForProject', () => {
+    let pageSize: number;
+    let paginationToken: string | undefined;
+
+    beforeEach(() => {
+      pageSize = 1;
+      paginationToken = '';
+    });
+
+    describe('when user role contains ITAdmin', () => {
+      beforeEach(() => {
+        mockUser = {
+          id: 'sampleId',
+          roles: ['ITAdmin']
+        };
+      });
+
+      describe('role only has ITAdmin', () => {
+        test('it succeeds, and response with empty data list and undefined pagination', async () => {
+          // BUILD
+          const mockResponse = {
+            data: [],
+            paginationToken: undefined
+          };
+          // OPERATE
+          const actualResponse = await dataSetService.listDataSetsForProject(
+            projectId,
+            mockUser,
+            pageSize,
+            paginationToken
+          );
+          // CHECK
+          expect(actualResponse).toEqual(mockResponse);
+        });
+      });
+
+      describe('role contains ITAdmin and ProjectAdmin', () => {
```

Comment:
:tada: great idea for a test. i don't think we've got many multi-role ones.

Reply:
Doesn’t hurt to test. The role is a list indicating we could have multiple roles, right?

Reply:
Someone be a pa for one proj and researcher for another